clap_derive/tests/generic.rs

```diff
@@ -0,0 +1,36 @@
+mod utils;
+
+use clap::{Args, Parser};
+
+#[test]
+fn generic() {
```

Comment:
Can you port the structopt test cases over?   They are covering more cases and it will help us identify user-facing impact of any differences.